CONFIGURATION.md

```diff
@@ -18,6 +18,7 @@ metadata.max.age.ms                      |  *  | 1 .. 86400000   |        900000
 topic.metadata.refresh.fast.interval.ms  |  *  | 1 .. 60000      |           250 | low        | When a topic loses its leader a new metadata request will be enqueued with this initial interval, exponentially increasing until the topic metadata has been refreshed. This is used to recover quickly from transitioning leader brokers. <br>*Type: integer*
 topic.metadata.refresh.fast.cnt          |  *  | 0 .. 1000       |            10 | low        | **DEPRECATED** No longer used. <br>*Type: integer*
 topic.metadata.refresh.sparse            |  *  | true, false     |          true | low        | Sparse metadata requests (consumes less network bandwidth) <br>*Type: boolean*
+topic.metadata.propagation.max.ms        |  *  | 0 .. 3600000    |         30000 | low        | Apache Kafka topic creation is asynchronous and it takes some time for a new topic to propagate throughout the cluster to all brokers. If a client requests topic metadata after manual topic creation but before the topic has been fully propagated to the broker the client is requesting metadata from, the topic will seem to be non-existent and the client will mark the topic as such, failing queued produced messages with `ERR__UNKNOWN_TOPIC`. This setting delays marking a topic as non-existent until the configured propagation max time has passed. The maximum propagation time is calculated from the time the topic is first referenced in the client, e.g., on produce(). <br>*Type: integer*
```

Comment:
does the java client have this configuration property, and if not, how does it avoid having it?

Reply:
The Java producer send() will block on initial metadata fetch and has a max.block.ms that controls how long non-blocking APIs (like send()) may block.

This is not the path we want to go in librdkafka, asynchronous APIs should be 100% asynchronous., and we already have asynchronous signaling in place (dr_cb).

Reply:
👍 absolutely can not block otherwise async calls, that'll totally screw up the api.